MapsetVerifier.Checks/Taiko/Design/CheckBgOffsetConsistency.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System.Numerics;
+using MapsetVerifier.Framework.Objects;
+using MapsetVerifier.Framework.Objects.Attributes;
+using MapsetVerifier.Framework.Objects.Metadata;
+using MapsetVerifier.Parser.Objects;
+
+namespace MapsetVerifier.Checks.Taiko.Design
+{
+    [Check]
+    public class CheckBgOffsetConsistency : GeneralCheck
+    {
+        private const string Minor = nameof(Issue.Level.Minor);
+        
+        public override CheckMetadata GetMetadata() =>
+            new BeatmapCheckMetadata()
+            {
+                Author = "Nostril",
+                Category = "Design",
+                Message = "Background offset inconsistencies",
+                Documentation = new Dictionary<string, string>()
+                {
+                    {
+                        "Purpose",
+                        @"
+                    Pointing out background offset inconsistencies between osu!taikodifficulties."
+                    },
+                    {
+                        "Reasoning",
+                        @"
+                    Background offset should generally be consistent across osu!taiko difficulties that share the same background."
+                    }
+                }
+            };
+
+        public override Dictionary<string, IssueTemplate> GetTemplates() =>
+            new()
+            {
+                {
+                    Minor,
+                    new IssueTemplate(Issue.Level.Minor,
+                        "\"{0}\" {1}: ({2})",
+                        "Filename",
+                        "Offset Coordinates",
+                        "List of Difficulties")
+                    .WithCause("Background offset is inconsistent across difficulties. Make sure this is intentional.")
+                }
+            };
+
+        public override IEnumerable<Issue> GetIssues(BeatmapSet beatmapSet)
+        {
+            // Record known offsets for each unique BG file
+            var files = new Dictionary<string, Dictionary<Vector2?, HashSet<string>>>();
+            
+            // Filter to osu!taiko beatmaps only
+            var taikoBeatmaps = beatmapSet.Beatmaps.Where(beatmap => beatmap.GeneralSettings.mode == Beatmap.Mode.Taiko).ToList();
```

Comment:
Instead of doing this, you can just add `Modes = [Beatmap.Mode.Taiko],` where the check is instantiated, under `Category = "Design",`

Reply:
Hmm, thats odd. Afaik, checks inheriting from `BeatmapCheck` should filter per-diff, while checks inheriting from `BeatmapSetCheck` or `GeneralCheck` are the ones which apply broadly to the whole set.

Might be worth taking another look to see if something else is funky. But if it still seems to be needed I guess just keep it.